### PR TITLE
Feat/updating component tests

### PR DESCRIPTION
**Pull request by shaal**
## Description

During JS guild, we added various new tests to `outline-button`
- [x] @shaal - `<button>` should always have a type (`button`/`submit`/`reset`), and by default set `type="button"`
- [x] @jirwincline
- [x] @LeeMellon
- [x] @peterschuelke 
- [x] @danmuzyka 

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Visual Testing
- [x] Automated Testing
- [ ] Accessibility Testing

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

<a href="https://gitpod.io/#https://github.com/phase2/outline/pull/314"></a>

### CODE REVIEW

File: src/components/base/outline-button/test/outline-button.test.ts [hunk 1]
```diff
@@ -1,7 +1,12 @@
 import { OutlineButton } from '../outline-button';
-import { assert, fixture, html } from '@open-wc/testing';
+import { assert, expect, fixture, html } from '@open-wc/testing';
+import { ButtonVariant, ButtonSize, ButtonType } from '../outline-button';
```
Comment (danmuzyka) [inline src/components/base/outline-button/test/outline-button.test.ts]: Couldn't we import these as part of line 1?

Comment (peterschuelke) [reply src/components/base/outline-button/test/outline-button.test.ts]: yep...

File: src/internal/isInType.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+//
```
Comment (peterschuelke) [inline src/internal/isInType.ts:1]: I debated on placing this file in `utilities`

File: src/components/base/outline-button/outline-button.ts [hunk 3]
```diff
@@ -51,7 +55,16 @@ export class OutlineButton extends OutlineElement {
   /**
    * The button style variant to use.
    */
-  @property({ type: String, attribute: 'button-variant' })
+  @property({
+    type: String,
+    attribute: 'button-variant',
+    converter: value => {
```
Comment (peterschuelke) [inline src/components/base/outline-button/outline-button.ts:78]: Adding a property converter to check if the provided value is in our custom type

**Pull request closed without merging**